pandas/plotting/_converter.py

```diff
@@ -178,6 +180,16 @@ class DatetimeConverter(dates.DateConverter):
 
     @staticmethod
     def convert(values, unit, axis):
+        # values might be a 1-d array, or a list-like of arrays.
+        if is_list_like(values) and is_list_like(values[0]):
```

Comment:
That's roughly what I ended up doing. One issue is that I don't want this accidentally consuming a generator, so if the outer container doesn't define `len` we return `False`. Will push in a few minutes.